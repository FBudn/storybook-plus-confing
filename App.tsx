import React from "react";
import Constants from "expo-constants";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { Provider } from "react-redux";
import LoginScreen from "./components/Screens/LoginScreen/LoginScreen";
import SignUpScreen from "./components/Screens/SignUpScreen/SignUpScreen";
import { store } from "./components/redux/store";

const Stack = createNativeStackNavigator();

function App() {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="LoginTemplate">
          <Stack.Screen
            name="LoginTemplate"
            component={LoginScreen}
            options={{ headerShown: false }}
          />
          <Stack.Screen
            name="SignUpTemplate"
            component={SignUpScreen}
            options={{ headerShown: false }}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
}

let AppEntryPoint = App;

if (Constants.expoConfig?.extra?.storybookEnabled === "false") {
  AppEntryPoint = require("./.storybook").default;
}

export default AppEntryPoint;
