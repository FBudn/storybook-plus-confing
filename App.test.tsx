/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render, screen } from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";
import TestRenderer from "react-test-renderer";
import App from "./App";

test(`should App render`, () => {
  render(<App />);
  const AppTest = screen.getByTestId(`test-app-id`);
  expect(AppTest).toBeOnTheScreen();
});

test("Should match snapshot:", () => {
  const AppTest = TestRenderer.create(<App />).toJSON();
  expect(AppTest).toMatchSnapshot();
});

test(`test App have text content `, () => {
  render(<App />);
  const AppTest = screen.getByTestId(`test-app-id`);
  expect(AppTest).toBeOnTheScreen();
  expect(AppTest).toHaveTextContent(
    `Open up App.tsx to start working on your app!`
  );
});
