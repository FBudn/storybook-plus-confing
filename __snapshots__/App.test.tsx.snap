// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot: 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="test-app-id"
>
  <Text>
    Open up App.tsx to start working on your app!
  </Text>
</View>
`;
